### PR TITLE
zip packaged apps asynchronously

### PR DESCRIPTION
**Pull request by mykmelez**
We should do expensive operations asynchronously. Like the archiving of packaged app files. Here's a change that switches that archiving to nsIZipWriter's asynchronous API.

### CODE REVIEW

**Comment by mykmelez** [comment]
@rpl: how does this look to you? It seems to work and is a relatively straightforward change. I'm not sure throwing the status is the best way to recover from an error, though.

File: addon/lib/simulator.js [hunk 1]
```diff
@@ -858,7 +859,17 @@ function archiveDir(zipFile, dirToArchive) {
...
+  writer.processQueue({
+    onStartRequest: function onStartRequest(request, context) {},
+    onStopRequest: function onStopRequest(request, context, status) {
+      if (status == Cr.NS_OK) {
+        writer.close();
+        console.log("archived dir " + dirToArchive);
+        callback();
+      }
+      else {
+        throw status;
```
Comment (rpl) [inline addon/lib/simulator.js]: here we can use the same approach used in the updateApp callback and pass the error as an optional callback first parameter (and often used in the nodejs async code callbacks)

```
let error = ...
callback(error);
```

and let the callback to handle success/error cases (e.g. what/how to report the result to the user)

Comment (mykmelez) [inline addon/lib/simulator.js]: Nice pattern! I made this change.

**Comment by mykmelez** [comment]
@rpl: how does this look with the change to error reporting?

**Comment by rpl** [comment]
:+1: 
it looks good to me

**Comment by mykmelez** [comment]
@rpl: Ok, thanks, I've merged it!

By the way, in the future, when I ask you to look at something like this, and you're satisfied with it, feel free to simply merge it yourself! In such a situation, you are serving as the reviewer of my code, so once you've looked at it and are satisfied with it, that means it's ready to go into the tree.

**Pull request merged**